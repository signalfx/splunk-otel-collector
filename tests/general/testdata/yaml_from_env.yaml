config_sources:
  env:
    defaults:
      JAEGER_PROTOCOLS: "{ protocols: { grpc: , } }"
      OTLP_PROTOCOLS: "{ grpc: , }"

receivers:
  jaeger:
    ${env:JAEGER_PROTOCOLS}
  otlp:
    protocols:
      ${env:OTLP_PROTOCOLS}
  hostmetrics:
    collection_interval: 2s
    scrapers:
      memory:
processors:
  filter:
    metrics:
      include:
        match_type: strict
        metric_names:
          - system.memory.usage
  metricstransform:
    transforms: ${env:YAML}

exporters:
  otlp:
    endpoint: "${OTLP_ENDPOINT}"
    tls:
      insecure: true
service:
  pipelines:
    traces:
      receivers:
        - jaeger
      exporters:
        - otlp
    metrics:
      receivers:
        - hostmetrics
        - otlp
      processors:
        - filter
        - metricstransform
      exporters:
        - otlp
