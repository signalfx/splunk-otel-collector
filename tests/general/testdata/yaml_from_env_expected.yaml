resourceMetrics:
  - resource: {}
    schemaUrl: https://opentelemetry.io/schemas/1.9.0
    scopeMetrics:
      - metrics:
          - description: Bytes of memory in use.
            name: system.memory.usage
            sum:
              aggregationTemporality: 2
              dataPoints:
                - asInt: "225095680"
                  attributes:
                    - key: state
                      value:
                        stringValue: buffered
                    - key: yaml-from-env
                      value:
                        stringValue: value-from-env
                  startTimeUnixNano: "1000000"
                  timeUnixNano: "2000000"
                - asInt: "4354945024"
                  attributes:
                    - key: state
                      value:
                        stringValue: cached
                    - key: yaml-from-env
                      value:
                        stringValue: value-from-env
                  startTimeUnixNano: "1000000"
                  timeUnixNano: "2000000"
                - asInt: "3161239552"
                  attributes:
                    - key: state
                      value:
                        stringValue: free
                    - key: yaml-from-env
                      value:
                        stringValue: value-from-env
                  startTimeUnixNano: "1000000"
                  timeUnixNano: "2000000"
                - asInt: "250576896"
                  attributes:
                    - key: state
                      value:
                        stringValue: slab_reclaimable
                    - key: yaml-from-env
                      value:
                        stringValue: value-from-env
                  startTimeUnixNano: "1000000"
                  timeUnixNano: "2000000"
                - asInt: "70791168"
                  attributes:
                    - key: state
                      value:
                        stringValue: slab_unreclaimable
                    - key: yaml-from-env
                      value:
                        stringValue: value-from-env
                  startTimeUnixNano: "1000000"
                  timeUnixNano: "2000000"
                - asInt: "4792176640"
                  attributes:
                    - key: state
                      value:
                        stringValue: used
                    - key: yaml-from-env
                      value:
                        stringValue: value-from-env
                  startTimeUnixNano: "1000000"
                  timeUnixNano: "2000000"
            unit: By
        scope:
          name: otelcol/hostmetricsreceiver/memory
          version: v0.97.0
